packages/alfa-rules/src/sia-r10/rule.ts

```diff
@@ -76,105 +77,116 @@ function hasTokens(input: string): boolean {
   return input.trim() !== "" && input.split(/\s+/).length > 0;
 }
 
+/**
+ * {@link https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-detail-tokens}
+ */ 
 const isValidAutocomplete: Predicate<Attribute> = (autocomplete) => {
   const tokens = autocomplete.value.toLowerCase().trim().split(/\s+/);
 
   let i = 0;
-  let next = tokens[i++];
+  let next: string | undefined = tokens[i++];
 
   if (next === undefined) {
     return false;
   }
 
+  // 1. Optional
   if (next.startsWith("section-")) {
     next = tokens[i++];
   }
 
+  // 2. Optional
   if (next === "shipping" || next === "billing") {
     next = tokens[i++];
   }
-
-  let field: string | null = null;
-
-  switch (next) {
-    case "name":
-    case "honorific-prefix":
-    case "given-name":
-    case "additional-name":
-    case "family-name":
-    case "honorific-suffix":
-    case "nickname":
-    case "username":
-    case "new-password":
-    case "current-password":
-    case "organization-title":
-    case "organization":
-    case "street-address":
-    case "address-line1":
-    case "address-line2":
-    case "address-line3":
-    case "address-level4":
-    case "address-level3":
-    case "address-level2":
-    case "address-level1":
-    case "country":
-    case "country-name":
-    case "postal-code":
-    case "cc-name":
-    case "cc-given-name":
-    case "cc-additional-name":
-    case "cc-family-name":
-    case "cc-number":
-    case "cc-exp":
-    case "cc-exp-month":
-    case "cc-exp-year":
-    case "cc-csc":
-    case "cc-type":
-    case "transaction-currency":
-    case "transaction-amount":
-    case "language":
-    case "bday":
-    case "bday-day":
-    case "bday-month":
-    case "bday-year":
-    case "sex":
-    case "url":
-    case "photo":
-      field = next;
-      break;
-
-    default:
-      switch (next) {
-        case "home":
-        case "work":
-        case "mobile":
-        case "fax":
-        case "pager":
-          next = tokens[i++];
-      }
-
-      switch (next) {
-        case "tel":
-        case "tel-country-code":
-        case "tel-national":
-        case "tel-area-code":
-        case "tel-local":
-        case "tel-local-prefix":
-        case "tel-local-suffix":
-        case "tel-extension":
-        case "email":
-        case "impp":
-          field = next;
-      }
+  
+  if (next === undefined) {
+    return true;
   }
 
-  if (field === null) {
-    return false;
+  // 3. Either of the following two options:
+  if (unmodifiableFields.includes(next)) {
+    next = tokens[i++];
+  } else if (modifiers.includes(next)) {
+    next = tokens[i++];
+    // Optional, but must be followed by:
+    if (modifiableFields.includes(next)) {
+      next = tokens[i++];
```

Comment:
Looks like the modifier is here mandatory before a modifiable field, but shouldn't.
The grammar here should be `unmodifiable | [modifier]? modifiable`, but the code seems to accept `unmodifiable | modifier modifiable`.